variables.tf

```
@@ -64,6 +64,12 @@ variable "authorized_ips" {
   default     = []
 }
 
+variable "public_network_access_enabled" {
+  description = "Whether enable public access for the App Service."
+  type        = bool
+  default     = false
```

This is a very impacting change. I'd keep the function app public by default.

```suggestion
  default     = true
```